libs/wingsdk/src/ui/section.ts

```diff
@@ -0,0 +1,106 @@
+import { Construct } from "constructs";
+import { VisualComponent } from "./base";
+import { Button, IButtonHandler } from "./button";
+import { Field, FieldProps, IFieldHandler } from "./field";
+import { fqnForType } from "../constants";
+import { App, UIComponent } from "../core";
+
+/**
+ * Global identifier for `Section`.
+ */
+export const SECTION_FQN = fqnForType("ui.Section");
+
+/**
+ * Props for `Section`.
+ */
+export interface SectionProps {
+  /**
+   * The label of the section.
+   * @default - no label
+   */
+  readonly label?: string;
+}
+
+/**
+ * A section can be used to group other visual components.
+ */
+export class Section extends VisualComponent {
+  /**
+   * Creates a new ui.Section instance through the app.
+   * @internal
+   */
+  public static _newSection(
+    scope: Construct,
+    id: string,
+    props: SectionProps = {}
+  ): Section {
+    return App.of(scope).newAbstract(SECTION_FQN, scope, id, props);
+  }
+
+  private readonly label: string | undefined;
+  private readonly subcomponents: VisualComponent[] = [];
+  private buttonIdx: number;
+  private fieldIdx: number;
+
+  constructor(scope: Construct, id: string, props: SectionProps = {}) {
+    super(scope, id);
+    this.label = props.label;
+    this.buttonIdx = 1;
+    this.fieldIdx = 1;
+  }
+
+  /**
+   * Adds a visual component to the section. The components
+   * will be rendered in the order they were added.
+   */
+  public add(...components: Array<VisualComponent>): void {
+    for (const component of components) {
+      if (component._newParent !== undefined) {
+        throw new Error(
+          `Cannot add ${component.node.path} to ${this.node.path} because it already belongs to ${component._newParent}.`
+        );
+      }
```

Comment:
this is a little hack. Basically, I thought it be nice to avoid boilerplate and allow folks to define buttons and labels and other stuff at a top level, in case they don't want to use `Section`:

```js
let button1 = new ui.Button();
let button2 = new ui.Button();
```

But the issue is if you do use a section:

```js
let section = new ui.Section();
section.add(button1, button2);
```

then "button1" and "button2" would now get rendered in both places (as a child of Section, and at the top level). To work around this, I added a field to keep track of whether each component was added to a grouping component.